go/integrityblock/integrityblock_test.go

```diff
@@ -193,3 +195,120 @@ func TestSignedWebBundleHasIntegrityBlockestAnother(t *testing.T) {
+	}
+
+	publicKeyAsReadableCborString, err := BytesToCborAndToReadableStringHelper(pub)
+	if err != nil {
+		t.Errorf("integrityBlock.CborBytes. err: %v", err)
+	}
+
+	signatureAsReadableCborString, err := BytesToCborAndToReadableStringHelper(integrityBlock.SignatureStack[0].Signature)
+	if err != nil {
+		t.Errorf("integrityBlock.CborBytes. err: %v", err)
+	}
+
+	want := `["🖋📦" "1b\x00\x00" [[map["ed25519PublicKey":` + publicKeyAsReadableCborString + `] ` + signatureAsReadableCborString + `]]]`
+
+	got, err := testhelper.CborBinaryToReadableString(integrityBlockBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if got != want {
+		t.Errorf("integrityblock: got: %s\nwant: %s", got, want)
+	}
+}
+
+// Test GetIntegrityBlockWithNewSignature when there is an existing signature
+// already in the stack. Signature attributes should always be "freshly" made.
+func TestGetIntegrityBlockWithExistingSignature(t *testing.T) {
+	pub, priv, err := ed25519.GenerateKey(rand.Reader)
+	if err != nil {
+		t.Error("Failed to generate test keys")
+	}
+
+	bundleFile, err := os.Open("./testfile.wbn")
+	if err != nil {
+		t.Error("Failed to open the test file")
+	}
+	defer bundleFile.Close()
+
+	webBundleHash, err := ComputeWebBundleSha512(bundleFile, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	integrityBlock := generateEmptyIntegrityBlock()
+	integrityBlock, err = GetIntegrityBlockWithNewSignature(integrityBlock, priv, webBundleHash, map[string][]byte{"hello": []byte("world")})
+	integrityBlock, err = GetIntegrityBlockWithNewSignature(integrityBlock, priv, webBundleHash, map[string][]byte{})
+
+	integrityBlockBytes, err := integrityBlock.CborBytes()
+	if err != nil {
+		t.Errorf("integrityBlock.CborBytes. err: %v", err)
+	}
+
+	publicKeyAsReadableCborString, err := BytesToCborAndToReadableStringHelper(pub)
+	if err != nil {
+		t.Errorf("integrityBlock.CborBytes. err: %v", err)
+	}
+
+	signatureAsReadableCborString1, err := BytesToCborAndToReadableStringHelper(integrityBlock.SignatureStack[0].Signature)
+	if err != nil {
+		t.Errorf("integrityBlock.CborBytes. err: %v", err)
+	}
+	signatureAsReadableCborString2, err := BytesToCborAndToReadableStringHelper(integrityBlock.SignatureStack[1].Signature)
+	if err != nil {
+		t.Errorf("integrityBlock.CborBytes. err: %v", err)
+	}
+
+	integritySignatureObject1 := `[map["ed25519PublicKey":` + publicKeyAsReadableCborString + `] ` + signatureAsReadableCborString1 + `]`
+	integritySignatureObject2 := `[map["ed25519PublicKey":` + publicKeyAsReadableCborString + ` "hello":"world"] ` + signatureAsReadableCborString2 + `]`
+	want := `["🖋📦" "1b\x00\x00" [` + integritySignatureObject1 + ` ` + integritySignatureObject2 + `]]`
+
+	got, err := testhelper.CborBinaryToReadableString(integrityBlockBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if got != want {
+		t.Errorf("integrityblock: got: %s\nwant: %s", got, want)
+	}
+}
+
+func BytesToCborAndToReadableStringHelper(bts []byte) (string, error) {
```

Comment:
Let's not export this function.
```suggestion
func bytesToCborAndToReadableStringHelper(bts []byte) (string, error) {
```